src/pkg/util/auth/token.go

```diff
@@ -0,0 +1,47 @@
+/*
+  Copyright (c) 2018, Sylabs, Inc. All rights reserved.
+
+  This software is licensed under a 3-clause BSD license.  Please
+  consult LICENSE file distributed with the sources of this project regarding
+  your rights to use or distribute this software.
+*/
+
+package auth
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// ReadToken reads a sylabs JWT auth token from a file
+func ReadToken(tokenPath string) string {
+	buf, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		sylog.Warningf("Couldn't read your Sylabs authentication token. Only pulls of public images will succeed.\n")
```

Comment:
We probably need to rethink these warnings now that they are happening on all init of the CLI. I think we should not warn if there is no token... unless we are using build --remote / push / pull things that may need a token.